web/src/components/Beneficiaries/BeneficiariesAddModal.tsx

```diff
@@ -115,13 +117,27 @@ const BeneficiariesAddModalComponent: React.FC<Props> = (props: Props) => {
         setCode2FA('');
     }, []);
 
+    const currencyPayloadAddress = React.useMemo(() => {
+        if (coinDestinationTag) {
+            if (isDestinationTagExists) {
+                return `${coinAddress}?dt=${coinDestinationTag}`;
+            }
+
+            if (isMemoTagExists) {
+                return `${coinAddress}?memo=${coinDestinationTag}`;
+            }
+        } else {
+            return coinAddress;
+        }
```

Comment:
```suggestion
        const address = requiresTag(currency) && coinDestinationTag ? `${rid}?dt=${tagMemo}`
            : requiresMemo(currency) && coinDestinationTag ? `${rid}?memo=${tagMemo}`
            : rid;
```

no need to make such a big function